app/controllers/application_controller.rb

```diff
@@ -73,7 +73,7 @@ def analytics
         sp: current_sp&.issuer,
         session: session,
         ahoy: ahoy,
-      )
+      ).decorate
```

Comment:
so if we're calling `.decorate` in the ApplicationController... we're decorating it basically everywhere (like 95% of callsites), which tells me we should probably just add whatever behavior here to the base class

Reply:
Got something working in 38235a73d.

Reply:
(I also had an alternative approach which avoided it being a "concern", but unsure how I feel about referencing the base event method class like this)

```rb
DECORATED_METHODS.each do |method_name|
  define_method(method_name) do |**kwargs|
    original_method = AnalyticsEvents.instance_method(method_name)
    original_method.bind_call(self, **kwargs, **common_analytics_attributes)
  end
end
```